### PR TITLE
Replace ui components with ui kit ones

### PR DESCRIPTION
**Pull request by bmigirl**

### CODE REVIEW

File: apps/checkout/src/components/Label.tsx [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
-  className,
-}) => (
-  <label
-    htmlFor={htmlFor}
-    className={clsx("text-text-primary text-base font-regular", className)}
-  >
-    {children}
-  </label>
+export const Label: React.FC<LabelProps> = ({ ...rest }) => (
+  <UiKitLabel {...rest} />
```
Comment (orzechdev) [inline apps/checkout/src/components/Label.tsx:12]: `htmlFor` is defined explicitly in props but all props are spread, I guess it's wrong

File: apps/checkout/src/components/RadioBoxGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { PropsWithChildren } from "react";
+
+interface RadioBoxGroupProps {
+  label: string;
+}
+
+export const RadioBoxGroup: React.FC<PropsWithChildren<RadioBoxGroupProps>> = ({
+  label,
```
Comment (orzechdev) [inline apps/checkout/src/components/RadioBoxGroup.tsx:8]: it's not used

File: apps/checkout/src/components/Label.tsx (hunk 1 above)
Comment (bmigirl) [reply apps/checkout/src/components/Label.tsx:12]: The thing here is that we want checkout to be accessible, hence we want to make sure we add this prop to all labels, but not in the ui kit which is to be used also in storefronts, so there we don't want to enforce that practice

File: apps/checkout/src/components/RadioBoxGroup.tsx (hunk 2 above)
Comment (bmigirl) [reply apps/checkout/src/components/RadioBoxGroup.tsx:8]: True, added

**Review by orzechdev — APPROVED** [review]

File: packages/ui-kit/src/components/Select/Select.tsx [hunk 3]
```diff
@@ -28,20 +31,27 @@ export interface SelectProps {
...
   options,
   error,
   disabled,
   classNames,
   onChange,
-}) => {
+}: SelectProps<TData>) => {
+  const selectedOption: Option<TData> | undefined = options.find(
+    ({ value }) => value === selectedValue
+  );
```
Comment (jwm0) [inline packages/ui-kit/src/components/Select/Select.tsx:47]: It looks like options is already typed so the type should be inferred:
```suggestion
  const selectedOption = options.find(
    ({ value }) => value === selectedValue
  );
```

File: apps/checkout/src/sections/AddressRadioBox.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+export const AddressRadioBox: React.FC<AddressRadioBoxProps> = ({
+  address,
+  onDelete,
+  onEdit,
+  ...rest
+}) => {
+  const formatMessage = useFormattedMessages();
+  const name = `${address.firstName} ${address.lastName}`;
+  const id = useRef(useId());
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/sections/AddressRadioBox.tsx:29]: If we leave it as useId it'll create a new Id on each rerender, but in fact we only need one per option so I'm thinking that useRef would be better here

File: apps/checkout/src/components/RadioBox/RadioBox.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  selectedValue: string | undefined;
+  subtitle?: string;
+}
+
+export const RadioBox: React.FC<RadioBoxProps> = ({ subtitle, ...rest }) => {
+  const radioProps = getRadioPropsFromRadioBoxProps(rest);
+
+  return (
+    <div className={clsx("radio-box", radioProps.checked && "selected")}>
+      <UiKitRadio {...radioProps} classNames={{ container: "!mb-0" }} />
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply apps/checkout/src/components/RadioBox/RadioBox.tsx:21]: Needs a bigger investigation, created a task for that

File: packages/ui-kit/src/components/Checkbox/Checkbox.stories.tsx [hunk 6]
```diff
@@ -8,18 +8,17 @@ export default {
...
-  useEffect(() => {
-    setEnabled(checked);
-  }, [checked]);
-
   return (
     <Checkbox
       {...args}
       checked={enabled}
-      onChange={(e) => setEnabled(e.currentTarget.checked)}
+      onChange={() => setEnabled(!enabled)}
```
[the remark this answers is not in the document]
Comment (bmigirl) [reply packages/ui-kit/src/components/Checkbox/Checkbox.stories.tsx:21]: Changed

File: apps/checkout/src/sections/AddressRadioBox.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jwm0) [reply apps/checkout/src/sections/AddressRadioBox.tsx:29]: useId does not create a new id on every render 🤔

**Review by szczecha — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**